### ISSUE TITLE
Request for a Spring ViewScope as in V8 for MPR based projects

### ISSUE THREAD

**Issue by StefanPenndorf** [vaadin/spring#508]
Ticket #263 contains the discussion about a ```@RouteScope``` that originated from the request of a ```@ViewScope``` for Vaadin Flow Spring Add-on. The discussion - as far as I understood - focuses on implementing a comparable scope for Vaadin Flow based (parts of) a project.

We're planning to migrate Vaadin 7 projects to Vaadin 14 using MPR with the "old" Vaadin 7 components for the vast majority of the views and functionality. The current projects heavily rely on ```@ViewScope``` when navigating between views.

Thus using Vaadin 14 with MPR and Vaadin Spring Add-on will not work for us due to the missing Scope.

Are there any plans to restore the old functionality in order to support such scenarios? Maybe using a special legacy package and warnings in JavaDoc that this scope will work with Vaadin 7 / Vaadin 8 navigator based navigation, only?

**Comment by pleku** [comment]
The plan is to introduce the `@RouteScope` as described in #263. Since there are nested routes in V10+ and not really just something that always maps to `@ViewScope`, we don't have any plans for adding back the `@ViewScope`. But in some context, the semantics of both should be the same as.

I'm closing this issue, please follow and +1 #263 instead, thanks.

**Comment by KyleRogers** [comment]
@pleku 
I don't think this is a duplicate. Maybe it's the wrong repository to post this issue at. The `@ViewScope` I requested is something that should be working with Navigator and MPR only. Maybe it needs to be added to MPR instead?

The ultimate goal is: If there is a Vaadin 7 Application that should be migrated to Vaadin 14 using MPR I'd like to have a replacement for `@ViewScope`. As far as I have investigates I would end up with tons of compile errors after upgrading to Vaadin 14 + MPR because there is no `@ViewScope` anymore. If I could use `@RouteScope` and this Scope will change when navigating to another View using Navigator than `@RouteScope` would be sufficient and I would replace `@ViewScope`  with `@RouteScope` for all my Vaadin 7 Views and corresponding beans.

BUT: As far as I read the discussion in #263 the `@RouteScope` will only apply to route changes using the new Router. But I might have misunderstood the discussion.

**Comment by pleku** [comment]
@KyleRogers thanks for the feedback, I see now what you mean.

We have to discuss on the technical semantics of this issue first, and then try to figure out where this could fit in the backlog. I'm not sure if it would help if we have `@RouteScope` available first or not.

I want to be honest and say that unless this is fairly trivial, it might take a while since I don't recall any other requests for this issue. If this is trivial, then probably we could look at this as we go about `@RouteScope`.

**Comment by pleku** [comment]
### Acceptance Criteria
- Add documentation under https://github.com/vaadin/docs/blob/latest/articles/tools/mpr/introduction/3-spring-boot.asciidoc#what-to-do-with-springview about migrating any views to use `MprRouteAdapter` and any `ViewScope` beans to use `@RouteScope`
- Add an IT that injects a `RouteScope` bean to a `MprRouteAdapter` based route target

**Comment by denis-anisimov** [comment]
The docs PR is not yet merged: accidentally closed , so reopened.
